File: samples/snippets/csharp/VS_Snippets_CLR_System/system.exception.class/cs/usageerrors1.cs
// <Snippet4>
using System;

public class Person
{
   private string _name;
   
   public string Name 
   {
      get { return _name; } 
      set { _name = value; }
   }
   
   public override int GetHashCode()
   {
      return this.Name.GetHashCode();  
   }  
                        
   public override bool Equals(object obj)
   {
      // This implementation contains an error in program logic:
      // It assumes that the obj argument is not null.
      Person p = (Person) obj;
      return this.Name.Equals(p.Name);
   }
}

public class Example
{
   public static void Main()
   {
      Person p1 = new Person();
      p1.Name = "John";
      Person p2 = null; 
      
      // The following throws a NullReferenceException.
      Console.WriteLine("p1 = p2: {0}", p1.Equals(p2));   
   }
}
// </Snippet4>

File: docs/fundamentals/runtime-libraries/snippets/System/Exception/Overview/vb/usageerrors1.vb
' Visual Basic .NET Document
Option Strict On

' <Snippet4>
Public Class Person
   Private _name As String
   
   Public Property Name As String
      Get
         Return _name
      End Get
      Set
         _name = value
      End Set
   End Property
   
   Public Overrides Function Equals(obj As Object) As Boolean
      ' This implementation contains an error in program logic:
      ' It assumes that the obj argument is not null.
      Dim p As Person = CType(obj, Person)
      Return Me.Name.Equals(p.Name)
   End Function
End Class

Module Example2
    Public Sub Main()
        Dim p1 As New Person()
        p1.Name = "John"
        Dim p2 As Person = Nothing

        ' The following throws a NullReferenceException.
        Console.WriteLine("p1 = p2: {0}", p1.Equals(p2))
    End Sub
End Module
' </Snippet4>
